.github/workflows/ci.yml

```diff
@@ -0,0 +1,77 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+env:
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Build Docker images
+        run: |
+          TARGET=prod make build
+          TARGET=dev make build
+
+      - name: Save Docker images
+        run: |
+          mkdir -p docker-image-prod docker-image-dev
+          docker save -o article-page-prod.tar libero/article-page:${IMAGE_TAG}
+          docker save -o article-page-dev.tar libero/article-page:${IMAGE_TAG}-dev
+
+      - name: Upload prod Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: article-page-prod
+          path: article-page-prod.tar
+
+      - name: Upload dev Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: article-page-dev
+          path: article-page-dev.tar
+
+  test:
+
+    needs: build
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        test:
+          - TARGET=dev .github/smoke-test.sh
+          - TARGET=prod .github/smoke-test.sh
+        include:
```

Comment:
the asymmetry between `dev` and `prod` here makes it difficult to understand what this section does. If we add:
```
- target: dev
  test: TARGET=dev .github/smoke-test.sh
```
it should keep working but also make explicit both steps rather than relying on a default for `matrix.target`?

Reply:
`include` doesn't work like Travis's `include`, in that it doesn't actually add in new jobs. https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-configurations-in-a-matrix-build

Reply:
How about https://github.com/libero/article-page/pull/1/commits/ad79c9a0cb38e7dfe0fe7876c431100f82895404? Means we'll have to `exclude` some things in the future (eg `prod` and `make lint`).